codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/JsonRpcProtocolGenerator.java

```diff
@@ -174,6 +176,9 @@ public void generateProtocolTests(GenerationContext context) {
     @Override
     protected void writeErrorMessageCodeDeserializer(GenerationContext context) {
         writeJsonErrorMessageCodeDeserializer(context);
+        if (context.getService().hasTrait(AwsQueryCompatibleTrait.class)) {
+            writeAwsQueryErrorCodeDeserializer(context);
+        }
     }
```

Comment:
From what I understand, this trait / behavior is only for the [AWS JSON 1.0 protocol](https://smithy.io/2.0/aws/protocols/aws-json-1_0-protocol.html).

If that's the case, the existing code in the generalized `JsonRpcProtocolGenerator` should not be touched, and instead be updated in [`AwsJsonRpc1_0`](https://github.com/aws/aws-sdk-go-v2/blob/main/codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AwsJsonRpc1_0.java) with a method override.